pylogit/base_multinomial_cm_v2.py

```diff
@@ -1230,7 +1234,7 @@ def _store_optional_parameters(self,
                                range(1, num_elements + 1)]
 
         # Store the names of the optional parameters in all_names
-        all_names = parameter_names + all_names
+        all_names = list(parameter_names) + list(all_names)
```

Comment:
Python 3.x is all about lazy evaluation and many well-know functions and methods no longer return lists. Instead they return _views_ or _iterators_, which makes concatenation a bit more complicated. For example following code won't work in Python 3.x:
```python
d = {1: 2}
# dict.keys() returns a view
d.keys() + [1, 2, 3]

# range behaves like xrange in Python 2.x
range(10) + [1, 2, 3]
```
There are many cases when _dict.keys()_ or _dict.values()_ are used so in order to insure that concatenation always works I explicitly convert  _parameter_names_ and _all_names_ to lists